mesonbuild/minstall.py

```diff
@@ -333,7 +337,13 @@ def remove(self, *args: T.Any, **kwargs: T.Any) -> None:
 
     def symlink(self, *args: T.Any, **kwargs: T.Any) -> None:
         if not self.dry_run:
-            os.symlink(*args, **kwargs)
+            try:
+                os.symlink(*args, **kwargs)
+            except (NotImplementedError, OSError):
```

Comment:
Since the symlink() function, like other os module wrappers, currently just directly wraps the real function, I think maybe it's a better idea to do this try/except in do_symlink instead.

Then, you could return False if symlinking fails, skip appending to the log, and skip setting self.did_install_something

Reply:
Makes sense. Now it looks quite obvious, sorry for the stupid mistake.

Reply:
No need to apologize :) a general rule in life and in computing is that most things make sense when you're blessed with hindsight. ;)